From a6740829d0de2e2d81c4c61892039770c1b04334 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Tue, 30 Aug 2022 17:49:04 +0300
Subject: [PATCH 37/42] ddr: Move code to separate makefile

Added s32_ddr.mk makefile which contains the DDR-related
settings previously in s32_common.mk.

Also removed the selection of platform-specific DDR files
(e.g. ddrc_cfg.c) from the platform-specific makefiles,
and relied on the S32_PLAT_SOC variable and on
PLAT_DDR_DRV and COMMON_DDR_DRV variables to include
the platform-specific and common DDR files, respectively.

Issue: ALB-9210
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_common.mk       | 12 +-----------
 plat/nxp/s32/s32_ddr.mk          | 24 ++++++++++++++++++++++++
 plat/nxp/s32/s32g/s32g2/s32g2.mk |  6 ------
 plat/nxp/s32/s32g/s32g3/s32g3.mk |  6 ------
 plat/nxp/s32/s32r/s32r.mk        |  6 ------
 5 files changed, 25 insertions(+), 29 deletions(-)
 create mode 100644 plat/nxp/s32/s32_ddr.mk

diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index 4ab0d8c3e..91c3836e6 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -9,6 +9,7 @@ include lib/libc/libc.mk
 include lib/libfdt/libfdt.mk
 include lib/xlat_tables_v2/xlat_tables.mk
 include make_helpers/build_macros.mk
+include plat/nxp/s32/s32_ddr.mk
 
 ERRATA_A53_855873	:= 1
 ERRATA_A53_836870	:= 1
@@ -33,17 +34,6 @@ $(eval $(call add_define_val,S32CC_EMU,$(S32CC_EMU)))
 S32GEN1_DRAM_INLINE_ECC	?= 1
 $(eval $(call add_define_val,S32GEN1_DRAM_INLINE_ECC,$(S32GEN1_DRAM_INLINE_ECC)))
 
-DDR_DRV = ${S32_DRIVERS}/ddr
-
-DDR_DRV_SRCS += \
-	${DDR_DRV}/ddr_density.c \
-	${DDR_DRV}/ddr_init.c \
-	${DDR_DRV}/ddr_utils.c \
-	${DDR_DRV}/ddr_lp.c \
-	${DDR_DRV}/ddr_lp_csr.c \
-	${DDR_DRV}/ddrss_cfg.c \
-	${DDR_DRV}/imem_cfg.c \
-
 BL2_AT_EL3		:= 1
 
 PLAT_INCLUDES 	+= \
diff --git a/plat/nxp/s32/s32_ddr.mk b/plat/nxp/s32/s32_ddr.mk
new file mode 100644
index 000000000..4bf2a60df
--- /dev/null
+++ b/plat/nxp/s32/s32_ddr.mk
@@ -0,0 +1,24 @@
+#
+# Copyright 2022 NXP
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+DDR_DRV = ${S32_DRIVERS}/ddr
+
+COMMON_DDR_DRV	= ${DDR_DRV}
+PLAT_DDR_DRV	= ${DDR_DRV}/${S32_PLAT_SOC}
+
+DDR_DRV_SRCS += \
+	${PLAT_DDR_DRV}/ddrc_cfg.c \
+	${PLAT_DDR_DRV}/dmem_cfg.c \
+	${PLAT_DDR_DRV}/dq_swap_cfg.c \
+	${PLAT_DDR_DRV}/phy_cfg.c \
+	${PLAT_DDR_DRV}/pie_cfg.c \
+	${COMMON_DDR_DRV}/ddr_init.c \
+	${COMMON_DDR_DRV}/ddr_utils.c \
+	${COMMON_DDR_DRV}/ddr_lp.c \
+	${COMMON_DDR_DRV}/ddr_lp_csr.c \
+	${COMMON_DDR_DRV}/ddrss_cfg.c \
+	${COMMON_DDR_DRV}/imem_cfg.c \
+	${DDR_DRV}/ddr_density.c \
diff --git a/plat/nxp/s32/s32g/s32g2/s32g2.mk b/plat/nxp/s32/s32g/s32g2/s32g2.mk
index 6c22d5d43..9bed6e6e3 100644
--- a/plat/nxp/s32/s32g/s32g2/s32g2.mk
+++ b/plat/nxp/s32/s32g/s32g2/s32g2.mk
@@ -6,12 +6,6 @@
 
 S32_PLAT_SOC := s32g2
 
-DDR_DRV_SRCS            += ${DDR_DRV}/s32g2/ddrc_cfg.c \
-			   ${DDR_DRV}/s32g2/dmem_cfg.c \
-			   ${DDR_DRV}/s32g2/dq_swap_cfg.c \
-			   ${DDR_DRV}/s32g2/phy_cfg.c \
-			   ${DDR_DRV}/s32g2/pie_cfg.c \
-
 include plat/nxp/s32/s32g/s32g_common.mk
 
 PLAT_SOC_PATH	:= ${S32_SOC_FAMILY}/${S32_PLAT_SOC}
diff --git a/plat/nxp/s32/s32g/s32g3/s32g3.mk b/plat/nxp/s32/s32g/s32g3/s32g3.mk
index 7c159cf69..6df418af6 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g3.mk
+++ b/plat/nxp/s32/s32g/s32g3/s32g3.mk
@@ -6,12 +6,6 @@
 
 S32_PLAT_SOC := s32g3
 
-DDR_DRV_SRCS            += ${DDR_DRV}/s32g3/ddrc_cfg.c \
-			   ${DDR_DRV}/s32g3/dmem_cfg.c \
-			   ${DDR_DRV}/s32g3/dq_swap_cfg.c \
-			   ${DDR_DRV}/s32g3/phy_cfg.c \
-			   ${DDR_DRV}/s32g3/pie_cfg.c \
-
 include plat/nxp/s32/s32g/s32g_common.mk
 
 PLAT_SOC_PATH	:= ${S32_SOC_FAMILY}/${S32_PLAT_SOC}
diff --git a/plat/nxp/s32/s32r/s32r.mk b/plat/nxp/s32/s32r/s32r.mk
index b0bf18540..aee418353 100644
--- a/plat/nxp/s32/s32r/s32r.mk
+++ b/plat/nxp/s32/s32r/s32r.mk
@@ -6,12 +6,6 @@
 
 S32_PLAT_SOC := s32r
 
-DDR_DRV_SRCS            += ${DDR_DRV}/s32r/ddrc_cfg.c \
-			   ${DDR_DRV}/s32r/dmem_cfg.c \
-			   ${DDR_DRV}/s32r/dq_swap_cfg.c \
-			   ${DDR_DRV}/s32r/phy_cfg.c \
-			   ${DDR_DRV}/s32r/pie_cfg.c \
-
 include plat/nxp/s32/s32_common.mk
 
 PLAT_SOC_PATH	:= ${S32_PLAT}/${S32_PLAT_SOC}
-- 
2.17.1

